### PR TITLE
Add PiP CFR experiment for 78

### PR DESCRIPTION
**Pull request by ncloudioj**
Yes, this experiment is targeting Firefox *78*.

Experimenter ticket: https://experimenter.services.mozilla.com/experiments/message-router-content-experiment-high-velocity-message-testing-cfr-to-pip/

### CODE REVIEW

File: messaging-experiments.json [hunk 1]
```diff
@@ -1146,5 +1146,488 @@
...
       ]
     }
+  },
+  {
+    "id": "MESSAGE_ROUTER_PICTURE_IN_PICTURE_CFR",
+    "enabled": true,
+    "filter_expression": "env.channel == 'release' && env.version >= '78.' && env.version < '79.'",
+    "targeting": "locale == 'en-US' && userPrefs.cfrFeatures &&\n'media.videocontrols.picture-in-picture.enabled'|preferenceValue &&\n[userId, 'messaging-experiments-cfr']|bucketSample(5312, 650, 10000) &&\n((currentDate|date - profileAgeCreated) / 86400000) > 7\n",
+    "arguments": {
+      "slug": "bug-1656382-message-recommend-picture-in-picture-cfr-release-77-80",
```
Comment (punamdahiya) [inline messaging-experiments.json:1156]: curious how we ensure slug above is same as slug from experimeneter.services.mozilla.com. I see that generated on click of Ready To Ship in Experimenter

Comment (ncloudioj) [reply messaging-experiments.json:1156]: No, we can't get the actual slug for now. This is merely a placeholder, and we have to update it once it's generated on Experimenter (and yes, that's very error-prone).

File: messaging-experiments.json [hunk 2]
```diff
@@ -1146,5 +1146,488 @@
         }
       ]
     }
+  },
+  {
+    "id": "MESSAGE_ROUTER_PICTURE_IN_PICTURE_CFR",
+    "enabled": true,
+    "filter_expression": "env.channel == 'release' && env.version >= '78.' && env.version < '79.'",
+    "targeting": "locale == 'en-US' && userPrefs.cfrFeatures &&\n'media.videocontrols.picture-in-picture.enabled'|preferenceValue &&\n[userId, 'messaging-experiments-cfr']|bucketSample(5312, 650, 10000) &&\n((currentDate|date - profileAgeCreated) / 86400000) > 7\n",
```
Comment (piatra) [inline messaging-experiments.json:1154]: Doesn't exclude people that opted out of studies `app.shield.optoutstudies.enabled`

File: messaging-experiments.json [hunk 3]
```diff
@@ -1146,5 +1146,488 @@
...
+                "sumo_path": "extensionrecommendations"
+              },
+              "layout": "icon_and_message",
+              "notification_text": {
+                "string_id": "cfr-doorhanger-extension-notification"
+              },
+              "text": "Keep yourself in the loop. With Picture-in-Picture you can watch the latest news while you visit other websites or apps - hover over a video and select the blue icon."
+            },
+            "frequency": {
+              "lifetime": 30,
```
Comment (piatra) [inline messaging-experiments.json:1241]: Is this correct? We want to show the message 30 times? The usual lifetime is `3`.

File: messaging-experiments.json [hunk 4]
```diff
@@ -1146,5 +1146,488 @@
...
+        {
+          "slug": "treatment-d",
+          "ratio": 1,
+          "groups": [
+            "cfr"
+          ],
+          "value": {
+            "id": "PIP_CFR_INFORMATIONAL",
+            "groups": [
+              "cfr"
```
Comment (piatra) [inline messaging-experiments.json:1497]: If we add the message to the `cfr` group then production messages can prevent experiment messages from being shown.

File: messaging-experiments.json (hunk 3 above)
Comment (ncloudioj) [reply messaging-experiments.json:1241]: Confirmed with experiment owners that we're good to use the lifetime as 3.

File: messaging-experiments.json (hunk 4 above)
Comment (ncloudioj) [reply messaging-experiments.json:1497]: Switching over to "cfr-experiments".

**Comment by ncloudioj** [comment]
@piatra FYI, the change highlights:

* Updated experiment slug
* Updated the branch slugs according to Experimenter
* Changed targeting version from 78 to 78+79
* Changed the sample size from 6.5% to 5%
* Added `platformName` to target Windows and Mac OS X

**Comment by piatra** [comment]
🚢

**Pull request merged**